File: CodeWarsJS/2kyuEvaluateMathematicalExpression/ReversePolishCalculator.js
"use strict";

module.exports = class ReversePolishCalculator {
    constructor() {
        this.calculatorUtilities = require("./CalculatorUtilities.js");
    }

    calculate(rpnTask) {
        if (rpnTask === "") return 0;

        let task = rpnTask.split(" ");
        let taskBuffer = [];

        task.forEach(taskLine => {
            if (taskLine === "") return;

            taskBuffer.push(
                !this.calculatorUtilities.isOperatorOrBracket(taskLine)
                    ? parseFloat(taskLine)
                    : this.calculatorUtilities.compute(taskLine, taskBuffer.pop(), taskBuffer.pop()));
        });

        return taskBuffer.pop();

    }
}

File: CodeWars/StringCalculator/ReversePolishCalculator.cs
using System.Collections.Generic;
using System.Linq;

namespace CodeWars.StringCalculator
{
    public class ReversePolishCalculator
    {
        public double Calculate(string rpnTask)
        {
            if (string.IsNullOrEmpty(rpnTask)) return 0;

            var task = new List<string>(rpnTask.Split(' '));
            var taskBuffer = new Stack<double>();

            foreach (var taskLine in task)
            {
                if (!string.IsNullOrEmpty(taskLine))
                {
                    taskBuffer.Push(
                    !CalculatorUtilities.IsOperatorOrBracket(taskLine)
                    ? double.Parse(taskLine)
                    : CalculatorUtilities.Compute(taskLine, taskBuffer.Pop(), taskBuffer.Pop())
                    );
                }

            }

            return taskBuffer.Count <= 1 ? taskBuffer.Pop() : taskBuffer.Sum();
        }
    }
}
